src/dashboard/store/modules/metamask.ts

```diff
@@ -92,20 +92,26 @@ export default {
       state.netId = -1;
     },
     addPendingDonation(state, donation) {
-      console.log('fucking lol donation');
-      console.log(donation);
       state.pendingDonations[donation.id] = donation;
       state.pendingDonations[donation.id].status = 'pending';
     },
     completePendingDonation(state, donation) {
-      console.log('completed donation!');
-      state.pendingDonations[donation.id].status = 'completed';
-      console.log(state.pendingDonations);
+      state.pendingDonations = {
+        ...state.pendingDonations,
+        [donation.id]: {
+          ...state.pendingDonations[donation.id],
+          status: 'completed',
+        },
+      };
```

Comment:
I find this a bit annoying to read, wouldn't `Vue.set()` be a lot cleaner?

Reply:
It would. Th problem is that Vue.set didn't trigger a UI update. I think the problem is that Vue tracks state.pendingDonations, but `Vue.set(state.pendingDonations[donation.id], 'status', 'completed')` changes the pendingDonations object but does not trigger changes in the UI. Probably
```
let updatedDonation = { ...state.pendingDonation[donation.id], status: 'completed' }
Vue.set(state.pendingDonations, donation.id, updatedDonation)
```
would work though. I'll try it

Reply:
For some reason that code did not trigger the reactivity. In the console the object is changed but the UI does not update. So I will keep it like this for now

Reply:
Fair enough